workloads/templates/pbench-worker-ds.yml.j2

```diff
@@ -62,6 +62,12 @@ spec:
         hostPath:
           path: /var/lib/kubelet/pods
       serviceAccountName: useroot
-      nodeSelector:
-        node-role.kubernetes.io/worker: ""
-        pbench_agent: "true"
+      affinity:
```

Comment:
@meowfaceman Thanks for the PR, we also need the logic to isolate the pbench-agent pods to only couple of nodes of interest since collecting the data from all the workers leads to duplication. We do it labeling the worker nodes of interest ( usually 2 nodes ) with pbench_agent=true and then isolate the pbench-agent worker daemonset to just those labeled nodes. The current logic deploys it on all the workers, we need to preserve the previous logic to avoid collecting data from all the worker nodes at large scale.

Reply:
Huh. So this is not the behavior that I was seeing prior to this change. Before this patch, on a cluster with 2 masters, 3 infra (also labeled workers), and 4 workers, the tooling workload tried to deploy 7 pbench workers. So that's with the code as is in the repo.

Reply:
I'll restore the pbench agent worker label though.